### PR TITLE
Fixing sphinx tests on windows

### PR DESCRIPTION
**Pull request by Andne**
This is for the issue #3592, I'm starting to look at the test individually and understand why they're failing.  Some are path issues, others appear to be related to other problems that need to be investigated.

I started by marking all failing tests as xfail, this makes it easier to debug one test at a time.  Now I'm going through and re-enabling tests so I can fix them.  I've opened this pull request early so that the changes can be reviewed as I go along if people desire.  Some of my changes are based on those attempted in #3619.

### CODE REVIEW

File: sphinx/ext/intersphinx.py [hunk 1]
```diff
@@ -112,18 +112,18 @@ def split_lines(iter):
             continue
         if location.endswith(u'$'):
             location = location[:-1] + name
-        location = join(uri, location)
+        location = posixpath.join(uri, location)
```
Comment (Andne) [inline sphinx/ext/intersphinx.py]: This came from PR #3619, there's a comment there about not using `posixpath.join` to join filesystem paths on windows. However it appears to me that this join is only used for generating the paths that will be injected into the output, which I think should always be posix/URL-type paths regardless of the OS and if the files are local or remote (the browser or other program should convert them back I think).

File: sphinx/ext/intersphinx.py [hunk 2]
```diff
@@ -224,8 +224,7 @@ def fetch_inventory(app, uri, inv):
                     uri = path.dirname(newinv)
         with f:
             try:
-                join = localuri and path.join or posixpath.join
-                invdata = read_inventory(f, uri, join)
+                invdata = read_inventory(f, uri)
```
Comment (Andne) [inline sphinx/ext/intersphinx.py:227]: This came from PR #3619, there's a comment there about not using `posixpath.join` to join filesystem paths on windows. However it appears to me that this `join` is only used for generating the paths that will be injected into the output, which I think should always be posix/URL-type paths regardless of the OS and if the files are local or remote (the browser or other program should convert them back I think).

**Comment by Andne** [comment]
I think I've fixed all of the tests that I can, the title underline test is still an expected failure, something about unicode doesn't work quite right during it, otherwise I was able to either fix code or fix tests to make everything else pass.

I'm a little unsure on commit a5b75f6, I don't have any python besides 2.7 available currently to test exactly what version it should work on, I'm just going off of the info in this python issue: http://bugs.python.org/issue9425

**Comment by Andne** [comment]
Accidentally introduced a style-check error for a long line, I've fixed that now too.

This build - https://ci.appveyor.com/project/Andne/sphinx/build/1.0.21 - shows the result of merging my appveyor PR and this PR together.

**Comment by tk0miya** [comment]
We merged #3867 instead finally. so this also should be updated or closed.
Could you check this please?

**Comment by Andne** [comment]
There's a few fixes for tests so they work properly here that weren't included in #3867 (that was more the minimal amount to run tests without failure and no code changes).  However, integration of 1.6 into stable causes so many conflicts that resolving them didn't seem possible any more and rebasing wasn't much better.

**Comment by tk0miya** [comment]
Thank you for quick response.
Please send another PR if you'd like.

**Pull request closed without merging**